### ISSUE TITLE
Keyboard focus escapes Modal when OK button is disabled

### ISSUE THREAD

**Issue by gillibrand** [carbon-design-system/carbon#7249]
## What package(s) are you using?

- [x] `carbon-components`
- [x] `carbon-components-react`

## Detailed description

When the primary button in a Modal is disabled, shift-tab will move the focus outside the dialog (i.e., the focus trap fails).
The issues appears to be that wrapFocus looks for the a focusable element in the dialog with a selector that includes `*[tabindex]:not([tabindex='-1'])`. And the Modal sets the primary button to `disabled="true"` but leaves `tabIndex=0`. That makes it look focusable to the selector, but the browser won't allow it (since it's disabled). So the call to .focus() fails, leaving focus on a sentinel element. The next shift-tab is then focused somewhere on the page outside the modal.

> Please create a reduced test case in CodeSandbox

See
https://codesandbox.io/s/gifted-panini-ow3pq
